### DISCUSSION TITLE
Need USA super chargers list to import into Teslamate

### DISCUSSION THREAD

**Discussion by rsunny2012** [teslamate-org/teslamate#3784] [Q&A] [answered]
Hello,

I do have a list which i have uploaded in 2018. Can somone help me here to get the latest list and how to import the data to my teslamate database

**Comment by parkr** [comment]
That's a neat idea. Is this just a list of geofences you can add? I find that teslamate identifies my supercharging pretty accurately. The only thing missing is cost, which varies based on time of day in my region.

**Reply by rsunny2012** [reply]
yes i am looking for geo fences list. it would be great if cost is also involved in that

**Reply by cwanja** [reply]
> it would be great if cost is also involved in that

Confused. Do you want cost or no? In a subsequent reply, you said you do not need cost.

**Reply by rsunny2012** [reply]
no i dont want cost information

**Comment by cwanja** [comment]
I see no technical limitations to this. But u am curious - do you really want to keep track of Tesla's varying costs as their costs change?

**Reply by rsunny2012** [reply]
Yes honestly we don’t need that data. I just want to update the super chargers info in us

**Comment by cwanja** [comment]
Can you find a way to extract the list from here? https://supercharge.info/changes

**Comment by init6-git** [comment]
I've sent the OP an SQL file that should contain all the SuCs in North America.
In case others want it:

[NASuC.txt](https://github.com/teslamate-org/teslamate/files/14765707/NASuC.txt)

**Comment by init6-git** [comment]
Here's the python script to get the info from supercharge.info. This was originally posted by PCmC & EVCarGuy on TMC. I just pulled them together.

https://teslamotorsclub.com/tmc/posts/8117226/

**Reply by init6-git** [reply]
And here's the code!`
#!/usr/bin/env python3

from urllib.request import urlopen
import json
url = "https://supercharge.info/service/supercharge/allSites"
response = urlopen(url)
sc_data = json.loads(response.read())

import datetime
stamp = datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')
for sc in sc_data:
 if sc['address']['region'] == "North America":
        name = sc['name']
        name = name.replace("'", "''")
        latitude = sc['gps']['latitude']
        longitude = sc['gps']['longitude']
        print(f"INSERT INTO public.geofences (name, latitude, longitude, radius, inserted_at, updated_at, cost_per_unit, session_fee, billing_type) VALUES ('{name} Supercharger', {latitude}, {longitude}, 35, '{stamp}', '{stamp}', 0.50, NULL, 'per_kwh');")

`

**Reply by rsunny2012** [reply, marked as answer]
i got the error. i have made slight modifications. below code works with python

#!/usr/bin/env python3

from urllib.request import urlopen
import json
import datetime

url = "https://supercharge.info/service/supercharge/allSites"
response = urlopen(url)
sc_data = json.loads(response.read())

stamp = datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S')
for sc in sc_data:
    if sc['address']['region'] == "North America":
        name = sc['name']
        name = name.replace("'", "''")
        latitude = sc['gps']['latitude']
        longitude = sc['gps']['longitude']
        print(f"INSERT INTO public.geofences (name, latitude, longitude, radius, inserted_at, updated_at, cost_per_unit, session_fee, billing_type) VALUES ('{name} Supercharger', {latitude}, {longitude}, 35, '{stamp}', '{stamp}', 0.50, NULL, 'per_kwh');")

**Reply by parkr** [reply]
This assumes 50 cents per kWh, right? It's a bummer that supercharge.info does not collect pricing.

**Reply by rsunny2012** [reply]
will any site gives us current super charging pricing where we can import the data ?

**Reply by init6-git** [reply]
Not that I'm aware of. Also as @parkr stated earlier some sites have different rates depending on time of day. Teslamate as it stands doesn't have the ability to have more than one price per geofence.
IIRC Tesla time-of-day pricing is not uniform. I'm pretty sure it is dependent on when a particular site is busy. So Supercharger A could have higher pricing between 4 and 6pm whereas Supercharger B could have higher pricing between 5pm and 7pm. Good luck coding that in Teslamate. :)

I don't think manually updating a geofence after I've visited a Supercharger to be a major overhead. How often do you charge at multiple different Superchargers in any one day? I reckon the updating of a Geofence table with a list would be more overhead.

**Reply by rsunny2012** [reply]
got it thanks

**Comment by brianmay** [comment]
We, as in the EV community really need a standard format to represent all the required data for every charger.

Plugshare perhaps comes closest to the existing solution. But I think they do not do a good job at capturing charging rates. And looks like they closely guard their data and don't make it freely available anyway. Even API access is restricted. https://company.plugshare.com/data.html https://developer.plugshare.com/access

But: creating a standard is easy. Getting people to use data in this standard and keep it up to date, that is the hard part.

Compulsory xkcd on the subject: https://xkcd.com/927/
